website/blog/2024-11-04-test-smarter-not-harder.md

```diff
@@ -0,0 +1,161 @@
+- *Completeness:* columns that should always contain text, *do.* Incomplete data is less useful!
+- *Formatting:* email addresses always have a valid domain. Incorrect emails may affect marketing outreach!
+
+### Bucket Two: Business-focused Anomalies
+
+*Business-focused anomalies* catch unexpected behavior. You can flag unexpected behavior by clearly defining *expected* behavior. *Business-focused anomalies* are when aspects of the data differ from what you know to be typical in your business. You’ll know what’s typical either through your own analyses, your colleagues’ analyses, or things your stakeholder homies point out to you.
+
+Since business-focused anomaly testing is set by a human, it will be fluid and need to be adjusted periodically. Here’s an example.
+
+Imagine you’re a sales analyst. Generally, you know that if your daily sales amount goes up or down by more than 20% daily, that’s bad. Specifically, it’s usually a warning sign for fraud or the order management system (OMS) dropping orders. You set a test in dbt to fail if any given day’s sales amount is a delta of 20% from the previous day. This works for a while.
+
+Then, you have a stretch of 3 months where your test fails 5 times a week! Every time you investigate, it turns out to be valid consumer behavior. You’re suddenly in hypergrowth, and sales are legitimately increasing that much.
+
+Your 20%-change fraud and OMS failure detector is no longer valid. You need to investigate anew which sales spikes or drops indicate fraud or OMS problems. Once you figure out a new threshold, you’ll go back and adjust your testing criteria.
+
+Although your data’s expected behavior will shift over time, you should still commit to defining business-focused anomalies to grow your understanding of what is normal for your data.
+
+Here’s how to identify potential anomalies.
+
+Start at your business intelligence (BI) layer. Pick 1-3 dashboards or tables that you *know* are used frequently. List these 1-3 dashboards or tables. For each dashboard or table you have, identify 1-3 “expected” behaviors that your end-users rely on.  Here are a few examples to get you thinking:
+
+- Revenue numbers should not change by more than X% in Y amount of time. This could indicate fraud or OMS problems.
+- Monthly active users should not decline more than X% after the initial onboarding period. This might indicate user dissatisfaction, usability issues, or that users not finding a feature valuable.
+- Exam passing rates should stay above Y%.  A decline below that threshold may indicate recent content changes or technical issues are affecting understanding or accessibility.
+
+You should also consider what data issues you have had in the past! Look through recent data incidents and pick out 3 or 4 to guard against next time. These might be in a #data-questions channel or perhaps a DM from a stakeholder.
+
+### Bucket 3: Stats-focused Anomalies
```

Comment:
```suggestion
### Bucket 3: Stats-focused anomalies
```

Reply:
I don't mind if you do One, Two, Three or 1, 2, 3 but you can't do One, Two, 3 😂

Reply:
dammit i really thought i had my numbers consistent 🫠  RIP!